urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/gui/MarketDisplay.java

```diff
@@ -0,0 +1,558 @@
+/**
+ * MarketDisplay.java
+ */
+package gui;
+
+import book.ProductService;
+import client.User;
+import java.awt.event.*;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.*;
+import javax.swing.table.DefaultTableCellRenderer;
+import javax.swing.table.DefaultTableModel;
+import price.Price;
+import price.PriceFactory;
+import publishers.MarketDataDTO;
+
+/**
+ * This class will handle displaying all the market data for the user.
+ *
+ * @author hieldc
+ * @version StockExVirtuoso.Version.1.0
+ */
+public class MarketDisplay extends javax.swing.JFrame {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 60 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.